src/roles/kserve/tests/create-devflag/main.sh

```
@@ -0,0 +1,116 @@
+  # Cases 1, 2, 3: Need to build and push image
+  # Handle container image building and pushing
+  info "Building image: ${REGISTRY_URL}/${COMPONENT_NAME}-controller:${CTRL_IMG_TAG}"
+  handle_image_build "$GIT_PARENT_REPO_DIR"
+  if [[ $? -ne 0 ]]; then
+    error "Failed to build image"
+    exit 1
+  fi
+  TARGET_IMAGE=${REGISTRY_URL}/${COMPONENT_NAME}-controller:${CTRL_IMG_TAG}
+fi
+
+cd $GIT_PARENT_REPO_DIR/${COMPONENT_NAME}
```

Shellcheck (0.10.0)

[warning] Use 'cd ... || exit' or 'cd ... || return' in case cd fails.

(SC2164)